### PR TITLE
Added support for custom headers in config

### PR DESCRIPTION
**Pull request by abhijitjaiswal**
Ability to add custom headers to rest call

### CODE REVIEW

**Comment by kevinykuo** [comment]
@abhijitjaiswal Thanks for the PR! Could you describe the use case(s) for this?

**Comment by abhijitjaiswal** [comment]
@kevinykuo The following is the use case:
Livy is running behind a web service that needs custom header to be passed for authentication before proxying request to Livy.

It is clean backward compatible change.

File: R/livy_connection.R [hunk 1]
```diff
@@ -48,7 +48,7 @@ livy_validate_http_response <- function(message, req) {
 #'
 #' @return Named list with configuration data
 livy_config <- function(config = spark_config(), username = NULL, password = NULL,
-                        csrf_header = TRUE) {
+                        custom_headers=NULL, csrf_header = TRUE) {
```
Comment (kevinykuo) [inline R/livy_connection.R]: Could we have spaces on either side of `=`?

File: R/livy_connection.R [hunk 2]
```diff
@@ -70,6 +70,12 @@ livy_config <- function(config = spark_config(), username = NULL, password = NUL
     )
   }
 
+  if(!is.null(custom_headers)){
```
Comment (kevinykuo) [inline R/livy_connection.R]: Could we have a space before each `{`?

**Comment by kevinykuo** [comment]
LGTM. @abhijitjaiswal Would you mind also adding a entry to `NEWS` under "Connections"?

@javierluraschi I think for simplicity we may want to get rid of `csrf_header` and default `custom_headers` to `list("X-Requested-By" = "sparklyr")`. If the user modifies the default, it'll still work with CSRF protection since any custom header gets through Livy's filter. We can do this in a separate PR.

**Comment by abhijitjaiswal** [comment]
@kevinykuo All suggestions are implemented, could you please merge the changes to master. Thanks for the quick responses 👍

**Comment by javierluraschi** [comment]
@kevinykuo your suggestion makes sense, merging this one then.

**Pull request merged**